vmm/src/device_manager.rs

```diff
@@ -3689,13 +3689,43 @@ impl DeviceManager {
 
     #[cfg(target_arch = "aarch64")]
     pub fn notify_power_button(&self) -> DeviceManagerResult<()> {
-        self.gpio_device
+        // There are three use cases:
+        // 1. The Cloud Hypervisor is built without feature acpi.
+        // 2. The Cloud Hypervisor is built with feature acpi, but users will
+        // use direct kernel boot with device tree.
+        // 3. The Cloud Hypervisor is built with feature acpi, and users will
+        // use ACPI+UEFI boot.
+        #[cfg(not(feature = "acpi"))]
+        // The `return` here will trigger a GPIO pin 3 event, which will trigger
+        // a power button event for use case 1.
+        return self
+            .gpio_device
             .as_ref()
             .unwrap()
             .lock()
             .unwrap()
             .trigger_key(3)
-            .map_err(DeviceManagerError::AArch64PowerButtonNotification)
+            .map_err(DeviceManagerError::AArch64PowerButtonNotification);
+        #[cfg(feature = "acpi")]
+        {
+            // Trigger a GPIO pin 3 event to satisify use case 2.
```

Comment:
I know triggering a GPIO event here is confusing, but I really cannot think up a better solution for use case 2. Please, any comments to improve this code are more than welcomed